File: corpus/dart-efficiency-design-v2/changed/lib/enrichment.dart
class Owner { final bool enabled; Owner(this.enabled); }
class Item { final bool active; final Owner? owner; Item(this.active, this.owner); }
abstract class Repository {
  Future<Item> load(String id);
  Future<List<Item>> loadMany(List<String> ids);
}

class ResultListFactory {
  static List<Item> create() => <Item>[];
}

Future<List<Item>> enrich(List<String> ids, Repository repo) async {
  final results = ResultListFactory.create();
  for (final id in ids) {
    final item = await repo.load(id);
    if (item.active) {
      if (item.owner != null) {
        if (item.owner!.enabled) results.add(item);
      }
    }
  }
  return results;
}

File: corpus/javascript-efficiency-design-v2/changed/enrichment.js
class ResultArrayFactory {
  static create() { return []; }
}

export async function enrich(ids, repo) {
  const results = ResultArrayFactory.create();
  for (const id of ids) {
    const item = await repo.load(id);
    if (item.active) {
      if (item.owner) {
        if (item.owner.enabled) results.push(item);
      }
    }
  }
  return results;
}
